lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp

```diff
@@ -134,24 +151,53 @@ void MmeNasStateConverter::guti_table_to_proto(
     } else {
       OAILOG_ERROR(LOG_MME_APP, "Key %s not in guti_ue_context_htbl", guti_str);
     }
+    OAILOG_DEBUG(
+        LOG_MME_APP, "guti_str:%s mme_ue_id:%d\n", guti_str.c_str(), mme_ue_id);
   }
   FREE_OBJ_HASHTABLE_KEY_ARRAY(key_array_p);
 }
 
+void MmeNasStateConverter::mme_app_convert_string_to_guti(
+    guti_t* guti_p, const std::string& guti_str) {
+  int idx         = 0;
+  std::size_t num = 1;
+#define HEX_BASE_VAL 16
+    guti_p->gummei.plmn.mcc_digit1 =
+        std::stoul(guti_str.substr(idx++, num), &num, HEX_BASE_VAL);
```

Comment:
Here is num is being used to determine the number of bytes to read in the substring, and also to point to the next character after the last read. This does not look right. Can we use different variables here?

Reply:
Yes, previously I had used two variables, but when viewed both had same value and getting updated at same time. So thought to use same variable.